src/main/java/frc/robot/SwerveModule.java

```diff
@@ -38,100 +69,112 @@ public SwerveModule(int moduleNumber, SwerveModuleConstants moduleConstants){
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kS = Constants.DriveCharacterization.Mod1.angleKS;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kV = Constants.DriveCharacterization.Mod1.angleKV;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kA = Constants.DriveCharacterization.Mod1.angleKA;
-
-            }
-            case 2 -> {
-                Robot.ctreConfigs.swerveDriveFXConfig.Slot0.kS = Constants.DriveCharacterization.Mod2.driveKS;
-                Robot.ctreConfigs.swerveDriveFXConfig.Slot0.kV = Constants.DriveCharacterization.Mod2.driveKV;
-                Robot.ctreConfigs.swerveDriveFXConfig.Slot0.kA = Constants.DriveCharacterization.Mod2.driveKA;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kS = Constants.DriveCharacterization.Mod2.angleKS;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kV = Constants.DriveCharacterization.Mod2.angleKV;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kA = Constants.DriveCharacterization.Mod2.angleKA;
-            }
-            case 3 -> {
-                Robot.ctreConfigs.swerveDriveFXConfig.Slot0.kS = Constants.DriveCharacterization.Mod3.driveKS;
-                Robot.ctreConfigs.swerveDriveFXConfig.Slot0.kV = Constants.DriveCharacterization.Mod3.driveKV;
-                Robot.ctreConfigs.swerveDriveFXConfig.Slot0.kA = Constants.DriveCharacterization.Mod3.driveKA;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kS = Constants.DriveCharacterization.Mod3.angleKS;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kV = Constants.DriveCharacterization.Mod3.angleKV;
-                Robot.ctreConfigs.swerveAngleFXConfig.Slot0.kA = Constants.DriveCharacterization.Mod3.angleKA;
-            }
-        }
+        driveMotor = new TalonFX(moduleConstants.driveMotorID);
+
+        Robot.ctreConfigs.swerveDriveFXConfig.Slot0 = Slot0Configs.from(moduleConstants.driveGains);
+        Robot.ctreConfigs.swerveAngleFXConfig.Slot0 = Slot0Configs.from(moduleConstants.angleGains);
         
-        mAngleMotor.getConfigurator().apply(Robot.ctreConfigs.swerveAngleFXConfig);
-        resetToAbsolute();
+        angleMotor.getConfigurator().apply(Robot.ctreConfigs.swerveAngleFXConfig);
+        driveMotor.getConfigurator().apply(Robot.ctreConfigs.swerveDriveFXConfig);
+        driveMotor.getConfigurator().setPosition(0.0);
 
-        mDriveMotor.getConfigurator().apply(Robot.ctreConfigs.swerveDriveFXConfig);
-        mDriveMotor.getConfigurator().setPosition(0.0);
+        drivePosition = driveMotor.getPosition();
+        driveVelocity = driveMotor.getVelocity();
+        anglePosition = angleMotor.getPosition();
+        angleEncoderPosition = angleEncoder.getAbsolutePosition();
 
+        driveMotorSim = driveMotor.getSimState();
+        angleMotorSim = angleMotor.getSimState();
+        angleEncoder.getSimState().setRawPosition(angleOffset.getRotations());
```

Comment:
i don't think this does anything in a real robot but we should probably put it in a if(IS_SIM) just in case/for clarity